classes/local/rule_checker.php

```diff
@@ -161,20 +239,30 @@ public function get_messages(): array {
         return $messages;
     }
 
+    /**
+     * Return checks' aggregated messages to be displayed as validation errors.
```

Comment:
Same here, maybe... "Return the aggregated messages from each check, to be displayed as validation errors."